src/pids.c

```diff
@@ -239,10 +243,122 @@ void decode_sis(pids_t *st, uint8_t *bits)
+            {
+                st->parameters[index] = parameter;
+                switch (index)
+                {
+                case 0:
+                    log_debug("Pending leap second offset: %d, current leap second offset: %d",
+                        parameter >> 8, parameter & 0xff);
+                    break;
+                case 1:
+                case 2:
+                    if (st->parameters[1] >= 0 && st->parameters[2] >= 0)
+                        log_debug("ALFN of pending leap second adjustment: %d", st->parameters[2] << 16 | st->parameters[1]);
+                    break;
+                case 3:
+                    tzo = (parameter >> 5) & 0x7ff;
+                    if (tzo > 1024) tzo -= 2048;
+                    log_debug("Local time zone offset: %d minutes, DST sched. %d, local DST? %s, regional DST? %s",
+                        tzo, (parameter >> 2) & 0x7, parameter & 0x2 ? "yes" : "no", parameter & 0x1 ? "yes" : "no");
+                    break;
+                case 4:
+                case 5:
+                case 6:
+                case 7:
+                    if (st->parameters[4] >= 0 && st->parameters[5] >= 0 && st->parameters[6] >= 0 && st->parameters[7] >= 0)
+                    {
+                        log_debug("Exciter manuf. \"%c%c\", core version %d.%d.%d.%d-%d, manuf. version %d.%d.%d.%d-%d",
+                            (st->parameters[4] >> 8) & 0x7f, st->parameters[4] & 0x7f,
+                            (st->parameters[5] >> 11) & 0x1f, (st->parameters[5] >> 6) & 0x1f, (st->parameters[5] >> 1) & 0x1f,
+                            (st->parameters[7] >> 11) & 0x1f, (st->parameters[7] >> 3) & 0x7,
+                            (st->parameters[6] >> 11) & 0x1f, (st->parameters[6] >> 6) & 0x1f, (st->parameters[6] >> 1) & 0x1f,
+                            (st->parameters[7] >> 6) & 0x1f, st->parameters[7] & 0x7
+                        );
+                    }
+                    break;
+                case 8:
+                case 9:
+                case 10:
+                case 11:
+                    if (st->parameters[8] >= 0 && st->parameters[9] >= 0 && st->parameters[10] >= 0 && st->parameters[11] >= 0)
+                    {
+                        log_debug("Importer manuf. \"%c%c\", core version %d.%d.%d.%d-%d, manuf. version %d.%d.%d.%d-%d",
+                            (st->parameters[8] >> 8) & 0x7f, st->parameters[8] & 0x7f,
+                            (st->parameters[9] >> 11) & 0x1f, (st->parameters[9] >> 6) & 0x1f, (st->parameters[9] >> 1) & 0x1f,
+                            (st->parameters[11] >> 11) & 0x1f, (st->parameters[11] >> 3) & 0x7,
+                            (st->parameters[10] >> 11) & 0x1f, (st->parameters[10] >> 6) & 0x1f, (st->parameters[10] >> 1) & 0x1f,
+                            (st->parameters[11] >> 6) & 0x1f, st->parameters[11] & 0x7
+                        );
+                    }
+                    break;
+                }
```

Comment:
Maybe add a default here with a log_debug message?

Reply:
The `index` is already checked above, so no other value is possible.

Reply:
Awesome.  I usually add a comment whenever I deliberately skip a default case in a switch statement.  Cool.